### ISSUE TITLE
Question: Best way to handle mix of command and non-command input?

### ISSUE THREAD

**Issue by TotallyInformation** [guidone/node-red-contrib-chatbot#90]
There does not seem to be a logical way to handle a bot that allows both commands and conversational input other than manually setting up a switch or function node to split the commands each to their own output with a catch-all output for whatever is left.

Is this the best way to approach this issue? If not, what is?

It would be REALLY helpful to have an example of this processing in the WIKI or your blog.

Here is an example of my code for handling this but it still isn't quite right because the switch generates a warning when the input isn't a command:

```
[{"id":"27c71ed5.d29e22","type":"comment","z":"432b859c.aafd0c","name":"","info":"You need this to add things to the variables\nthat RiveScript can pick up, it cannot read\ndirect from msg.","x":902.5,"y":323.6000061035156,"wires":[]},{"id":"714982d9.ad047c","type":"chatbot-telegram-receive","z":"432b859c.aafd0c","bot":"edbd32e.1e115d","x":250,"y":260,"wires":[["dc8928d9.aead88","231920cf.74394","7ba2e5d5.b9c90c"]]},{"id":"8c5da2c2.e92c7","type":"chatbot-message","z":"432b859c.aafd0c","name":"","message":[{"message":"Hello {{firstName}}, how are you today?"}],"answer":false,"track":false,"parse_mode":"","x":1210,"y":380,"wires":[["628692f7.b2b1ec"]]},{"id":"80f8d386.3950e","type":"chatbot-command","z":"432b859c.aafd0c","name":"","command":"/hi","x":930,"y":100,"wires":[["b2d56ed.ac6359"]]},{"id":"dc8928d9.aead88","type":"debug","z":"432b859c.aafd0c","name":"ORIGINAL","active":true,"console":"false","complete":"true","x":476.20001220703125,"y":202.1999969482422,"wires":[]},{"id":"231920cf.74394","type":"chatbot-debug","z":"432b859c.aafd0c","x":472.49998474121094,"y":165,"wires":[]},{"id":"628692f7.b2b1ec","type":"chatbot-telegram-send","z":"432b859c.aafd0c","bot":"edbd32e.1e115d","track":false,"outputs":0,"x":1581.9999542236328,"y":200,"wires":[]},{"id":"7ba2e5d5.b9c90c","type":"chatbot-authorized","z":"432b859c.aafd0c","x":470.4999694824219,"y":270.40000915527344,"wires":[["f37e816c.3528a"],["d31b007b.ef58d"]]},{"id":"d31b007b.ef58d","type":"chatbot-message","z":"432b859c.aafd0c","name":"","message":[{"message":"I'm sorry {{firstName}}. You are not authorised to use this bot. If you would like to and think you should be allowed, please contact me in real life."}],"answer":true,"track":false,"parse_mode":"","x":1150,"y":460,"wires":[["628692f7.b2b1ec"]]},{"id":"b30cb056.fe974","type":"chatbot-rivescript","z":"432b859c.aafd0c","name":"","script":"! version = 2.0\n\n// https://www.rivescript.com/docs/tutorial#getting-started\n\n+ (hi|hello)[ *]\n- Hello <get firstName>, my name is Pi BOT\n\n// Demonstrates both alternatives and an optional\n+ is it (day|night)[?]\n* <get daynight> == undefined => I don't know if it is day or night!.\n- >get firstName>, It is <get daynight>\n\n+ why did you say that[?]\n- I said, `<reply>`, because you said, `<input>`.\n\n+ what[?]\n@ why did you say that\n\n+ who said that[?]\n- Hi, this is Pi BOT here <get firstNamee>.\n\n+ *\n- What _are_ you talking about <get firstName>?","x":1043,"y":359,"wires":[["aa39e709.9e5278","6aaf58c0.e6cf88"],["8c5da2c2.e92c7"]]},{"id":"aa39e709.9e5278","type":"chatbot-message","z":"432b859c.aafd0c","name":"","message":[{"message":"{{payload}}"}],"answer":false,"track":false,"parse_mode":"Markdown","x":1210,"y":340,"wires":[["628692f7.b2b1ec"]]},{"id":"6aaf58c0.e6cf88","type":"chatbot-debug","z":"432b859c.aafd0c","x":1210,"y":300,"wires":[]},{"id":"2a69704d.d57fa","type":"function","z":"432b859c.aafd0c","name":"","func":"var home = global.get('home');\nconsole.warn(home);\nvar daynight = home.daynight;\nconsole.warn(daynight);\n\nvar chat = msg.chat();\nconsole.log(chat.get('authorized')); // is the user authorized\nconsole.log(chat.get('username')); // guidone72\nchat.set('daynight', daynight);\n\nreturn msg;","outputs":1,"noerr":0,"x":903,"y":359,"wires":[["b30cb056.fe974"]]},{"id":"9212bde4.94504","type":"chatbot-command","z":"432b859c.aafd0c","name":"","command":"/start","x":930,"y":140,"wires":[["bac9be65.6ad69"]]},{"id":"5cdd4e77.f6fe7","type":"chatbot-command","z":"432b859c.aafd0c","name":"","command":"/help","x":930,"y":220,"wires":[["1715f665.7a67ca"]]},{"id":"78020c96.5a1b14","type":"chatbot-command","z":"432b859c.aafd0c","name":"","command":"/settings","x":940,"y":260,"wires":[["e2dc35d0.70e218"]]},{"id":"b2d56ed.ac6359","type":"chatbot-message","z":"432b859c.aafd0c","name":"","message":[{"message":"Hello {{firstName}}, how are you today? ({{message}})"}],"answer":false,"track":false,"parse_mode":"","x":1130,"y":100,"wires":[["628692f7.b2b1ec"]]},{"id":"f37e816c.3528a","type":"switch","z":"432b859c.aafd0c","name":"Check input type","property":"originalMessage.entities[0].type","propertyType":"msg","rules":[{"t":"eq","v":"bot_command","vt":"str"},{"t":"else"}],"checkall":"true","outputs":2,"x":704.5,"y":256,"wires":[["80f8d386.3950e","9212bde4.94504","5cdd4e77.f6fe7","78020c96.5a1b14","c5962950.d9b238"],["2a69704d.d57fa"]],"outputLabels":["A Telegram Bot Command",null]},{"id":"bac9be65.6ad69","type":"chatbot-message","z":"432b859c.aafd0c","name":"","message":[{"message":"Thanks {{firstName}}, I am now connected to you ({{message}})"}],"answer":true,"track":false,"parse_mode":"","x":1130,"y":140,"wires":[["628692f7.b2b1ec"]]},{"id":"c5962950.d9b238","type":"chatbot-command","z":"432b859c.aafd0c","name":"","command":"/stop","x":930,"y":180,"wires":[["70be58cc.e20428"]]},{"id":"1715f665.7a67ca","type":"chatbot-message","z":"432b859c.aafd0c","name":"","message":[{"message":"Hi {{firstName}}, The following commands are available:\n/start - Start interacting with me\n/stop - You no longer care for me!\n/help - this text\n/settings - controls me\n/hi - Makes sure I am listening"}],"answer":true,"track":false,"parse_mode":"","x":1130,"y":220,"wires":[["628692f7.b2b1ec"]]},{"id":"70be58cc.e20428","type":"chatbot-message","z":"432b859c.aafd0c","name":"","message":[{"message":"Well {{firstName}}, I'm sorry to see you go. Let me know if you want to connect up again in the future."}],"answer":true,"track":false,"parse_mode":"","x":1130,"y":180,"wires":[["628692f7.b2b1ec"]]},{"id":"e2dc35d0.70e218","type":"chatbot-message","z":"432b859c.aafd0c","name":"","message":[{"message":"Hi {{firstName}}, Sorry, there are not any settings as yet."}],"answer":true,"track":false,"parse_mode":"","x":1130,"y":260,"wires":[["628692f7.b2b1ec"]]},{"id":"edbd32e.1e115d","type":"chatbot-telegram-node","z":"","botname":"MyTestBot","usernames":"","polling":"1000","log":""}]
```

**Comment by guidone** [comment]
@TotallyInformation I get your problem, my idea was to not process all command-like in all parse node (except `Command node` of course), so they will not trigger any default message if there's a no match

What do you think?

Thanks for the feedback

**Comment by guidone** [comment]
Fixed in 0.7.4

Thanks for the feedback
